Tests/SentryTests/SentryPredicateDescriptorTests.swift

```diff
@@ -0,0 +1,152 @@
+        assertPredicate(predicate: pred, expectedResult: "field1 CONTAINS %@")
+    }
+    
+    func test_betweenOperator() {
+        let pred = NSPredicate(format: "field1 between %@")
+        assertPredicate(predicate: pred, expectedResult: "field1 BETWEEN %@")
+    }
+    
+    func test_likeOperator() {
+        let pred = NSPredicate(format: "field1 like %@")
+        assertPredicate(predicate: pred, expectedResult: "field1 LIKE %@")
+    }
+    
+    func test_andCompound() {
+        let pred = NSPredicate(format: "field1 = %@ and field2 = %@")
+        assertPredicate(predicate: pred, expectedResult: "field1 == %@ AND field2 == %@")
+    }
+    
+    func test_orCompound() {
+        let pred = NSPredicate(format: "field1 = %@ or field2 = %@")
+        assertPredicate(predicate: pred, expectedResult: "field1 == %@ OR field2 == %@")
+    }
+    
+    func test_notCompound() {
+        let pred = NSPredicate(format: "not field1 = %@")
+        assertPredicate(predicate: pred, expectedResult: "NOT field1 == %@")
+    }
+    
+    func test_AggregateExpression() {
+        let pred = NSPredicate(format: "field1 in {1,2,3,4}")
+        assertPredicate(predicate: pred, expectedResult: "field1 IN {%@, %@, %@, %@}")
+    }
+    
+    func test_ternaryExpression() {
+        let pred = NSPredicate(format: "ternary(field1 = %@ , 1 , 2) == 1")
+        assertPredicate(predicate: pred, expectedResult: "TERNARY(field1 == %@,%@,%@) == %@")
+    }
+    
+    func test_compoundInCompound() {
+        let pred = NSPredicate(format: "field1 = 1 And field2 = 2 or field2 = 1")
+        assertPredicate(predicate: pred, expectedResult: "(field1 == %@ AND field2 == %@) OR field2 == %@")
+    }
+    
+    func test_compoundInCompound_2() {
+        let pred = NSPredicate(format: "field1 = 1 And (field3 = 2 or field2 = 1)")
+        assertPredicate(predicate: pred, expectedResult: "field1 == %@ AND (field3 == %@ OR field2 == %@)")
+    }
+    
+    func test_UNKNOWN() {
+        let pred = NSPredicate { _, _ in
+            return false
+        }
+        assertPredicate(predicate: pred, expectedResult: "<UNKNOWN>")
+    }
+    
+    func test_invalidCompound() {
+        guard let invalidCompound = NSCompoundPredicate.LogicalType(rawValue: 6) else {
+            XCTFail("Could not create invalid compound type")
+            return
+        }
+        
+        let pred = NSCompoundPredicate(type: invalidCompound, subpredicates: [NSComparisonPredicate(format: "field1 == 1"), NSComparisonPredicate(format: "field2 == 2")])
+        
+        assertPredicate(predicate: pred, expectedResult: "field1 == %@, field2 == %@")
+    }
+    
+    func test_invalidComparison() {
+        let pred = NSComparisonPredicate(leftExpression: NSExpression(format: "field1"), rightExpression: NSExpression(format: "1"), customSelector: #selector(compareFunction(_:_:)))
+        
+        assertPredicate(predicate: pred, expectedResult: "<COMPARISON NOT SUPPORTED>")
```

Comment:
`m`: I'm pretty sure that this could confuse users. Imagine the span description in Sentry
`SELECT 'TestEntity' WHERE field1 <COMPARISON NOT SUPPORTED> %@ SORT BY field1 DESCENDING`. 
Maybe `OPERATOR UNKOWN` would be better?

Reply:
All the expression will be omitted because we don't know how it was supposed to be displayed.
But we could use another expression.